### PR TITLE
Add status service for backup download

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13329 

This PR introduces `BackupDownloadStatusService` which will be used along with the existing `BackupDownloadProgressChecker` to manage the execution and state of preparing a backup to a downloadable file. This is base logic, there may be changes once the UI development is underway.

**Merge instructions**
- Ensure that PR #13425 has been merged
- Remove merge label
- Merge as normal

**To test:**
Feature is not being used yet, CI test success will be sufficient until UI is implemented.

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by malinajirka** [comment]
I've updated the target milestone to 16.4. @zwarm when you come back, could you please fix the build before we dive into the review. Thanks!

**Comment by zwarm** [comment]
@malinajirka  - thanks, It needed the latest develop merged in to pick up the FluxC changes.

File: WordPress/src/test/java/org/wordpress/android/ui/activitylog/BackupDownloadStatusServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+                backupDownloadStatusService.requestStatusUpdate()
+
+                verify(backupDownloadProgressChecker).startNow(
+                        site,
+                        inProgressBackupDownloadStatusModel.downloadId
+                )
+            }
+
+    @Test
+    fun `triggers BackupDownload And Makes Action Unavailable`() = runBlocking {
```
Comment (malinajirka) [inline WordPress/src/test/java/org/wordpress/android/ui/activitylog/BackupDownloadStatusServiceTest.kt]: Typo - capital letters: **A**nd **M**akes **A**ction **U**navailable

**Review by malinajirka — APPROVED** [review]
Thanks @zwarm!

It seems the code is consistent with the RewindStatusService and its tests. There are serveral things I'd have commented on, but since it's consistent with this closely related class I decided to skip the comments. We can do a small refactoring at the end of the project to make both classes a bit more consistent with our current code style.

File: WordPress/src/test/java/org/wordpress/android/ui/activitylog/BackupDownloadStatusServiceTest.kt (hunk 1 above)
Comment (zwarm) [reply WordPress/src/test/java/org/wordpress/android/ui/activitylog/BackupDownloadStatusServiceTest.kt]: Thanks - updated in d7d59e8

**Pull request merged**